lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java

```diff
@@ -0,0 +1,177 @@
+        File file = new File(fileUrl);
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (Exception e) {
+                Log.e(TAG, "Could not create file!");
+            }
+        }
+        try {
+            OutputStreamWriter fileWriter = new OutputStreamWriter(new FileOutputStream(file, true));
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+            String rawLog = log + " " + message;
+            bufferedWriter.write(rawLog);
+            bufferedWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private String getFileUrl() {
+        File file1 = new File(mLogFileUrl);
+        File file2 = new File(mLogFileUrl2);
+
+        boolean biggerFile1 = file1.length() > LOGS_SIZE_LIMIT;
+        boolean biggerFile2 = file2.length() > LOGS_SIZE_LIMIT;
+
+        if (biggerFile1 && biggerFile2) {
+            if (file1.exists()) {
+                file1.delete();
+            }
+            return mLogFileUrl;
+        }
+
+        if (biggerFile1) {
+            if (file2.exists()) {
+                file2.delete();
+            }
+            return mLogFileUrl2;
+        }
+
+        return mLogFileUrl;
+    }
+
+    public void d(@NotNull String tag, String message) {
+        saveLog(LogLevel.d, tag, message);
+        Log.d(tag, message);
+    }
+
+    public void d(@NotNull String tag) {
+        saveLog(LogLevel.d, tag, "");
+        Log.d(tag, "");
+    }
+
+    public void e(@NotNull String tag, String message) {
+        saveLog(LogLevel.e, tag, message);
+        Log.e(tag, message);
+    }
+
+    public void e(@NotNull String tag) {
+        saveLog(LogLevel.e, tag, "");
+        Log.e(tag, "");
+    }
+
+    public void i(@NotNull String tag, String message) {
+        saveLog(LogLevel.i, tag, message);
+        Log.i(tag, message);
+    }
+
+    public void i(@NotNull String tag) {
+        saveLog(LogLevel.i, tag, "");
+        Log.i(tag, "");
+    }
+
+    public void v(@NotNull String tag, String message) {
+        saveLog(LogLevel.v, tag, message);
+        Log.v(tag, message);
+    }
+
+    public void v(@NotNull String tag) {
+        saveLog(LogLevel.v, tag, "");
+        Log.v(tag, "");
+    }
+
+    public void w(@NotNull String tag, String message) {
+        saveLog(LogLevel.w, tag, message);
+        Log.w(tag, message);
+    }
+
+    public void w(@NotNull String tag) {
+        saveLog(LogLevel.w, tag, "");
+        Log.w(tag, "");
+    }
+
+    public void wtf(@NotNull String tag, String message) {
+        saveLog(LogLevel.wtf, tag, message);
+        Log.wtf(tag, message);
+    }
+
+    public void wtf(@NotNull String tag) {
+        saveLog(LogLevel.wtf, tag, "");
+        Log.wtf(tag, "");
+    }
+
+    private enum LogLevel {
+        d("LOG"),
+        e("ERROR"),
+        i("LOG"),
+        v("LOG"),
+        w("WARN"),
+        wtf("WTF");
```

Comment:
wtf?

Reply:
on android there's a log level called wtf for extreme cases